### PR TITLE
Allow configuration of required fields for company details form, add `publicEmail`

### PR DESCRIPTION
**Pull request by Shinsina**
**REQUIRES: https://github.com/parameter1/base-cms/pull/767 to be deployed to the GraphQL servers**

When `email` is required:
Screenshot from 2023-08-14 14-57-02
Screenshot from 2023-08-14 14-57-22
Screenshot from 2023-08-14 14-57-28
Screenshot from 2023-08-14 14-58-00

When no fields are required:
Screenshot from 2023-08-14 15-02-19

Presently only supports setting `email` to required but can be extended if we see a need for it "immediately" otherwise it should be a fairly modular thing to apply to other fields as need be.

### CODE REVIEW

**Comment by Shinsina** [comment]
[Email Address Change for Client Portal](https://trello.com/c/5M6Qk7ct/388-email-address-change-for-client-portal)

File: services/app/app/controllers/portal/company.js [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 
 export default Controller.extend({
+  config: inject(),
 
   isModalOpen: false,
 
+  submitDisabled: computed('config.requiredCompanyFields', 'model', function() {
+    const requiredFields = this.config.requiredCompanyFields || [];
+    const requiredValuesSet = requiredFields.map((fieldName) => this.get(`model.${fieldName}`)).filter((v) => v);
+    console.log(requiredFields, requiredValuesSet);
```
Comment (B77Mills) [inline services/app/app/controllers/portal/company.js]: @Shinsina  Remove me.

File: services/app/app/controllers/portal/company.js [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+import { inject } from '@ember/service';
 
 export default Controller.extend({
+  config: inject(),
 
   isModalOpen: false,
 
+  submitDisabled: computed('config.requiredCompanyFields', 'model', function() {
+    const requiredFields = this.config.requiredCompanyFields || [];
+    const requiredValuesSet = requiredFields.map((fieldName) => this.get(`model.${fieldName}`)).filter((v) => v);
```
Comment (B77Mills) [inline services/app/app/controllers/portal/company.js:13]: @Shinsina  I dont think we need the .filter part of this.

**Review by B77Mills — CHANGES_REQUESTED** [review]

File: services/app/app/templates/components/company-card/contact.hbs [hunk 3]
```diff
@@ -64,14 +64,34 @@
...
+      {{/if}}
+
       <div class="input-group">
         {{input
           type="email"
           class="form-control"
           id="email-input"
           aria-describedby="email-help"
           value=model.email
+          change=(action onChange)
```
Comment (solocommand) [inline services/app/app/templates/components/company-card/contact.hbs:94]: The input should also be `required` based on the `is-required` check.
```suggestion
          change=(action onChange)
          required=(is-required "email")
```

**Review by B77Mills — APPROVED** [review]

**Pull request merged**